data/constants/keycodes/extras/keycodes_farsi_0.0.1.hjson

```diff
@@ -0,0 +1,632 @@
+            "key": "FA_MMNS",
+            "label": "−",
+        }
+        "ALGR(FA_ZAD)": {
+            "key": "FA_DEG",
+            "label": "°",
+        }
+        "ALGR(FA_SE)": {
+            "key": "FA_EURO",
+            "label": "€",
+        }
+        "ALGR(FA_HE)": {
+            "key": "FA_LTRO",
+            "label": "‭",
+        }
+        "ALGR(FA_KHE)": {
+            "key": "FA_RTLO",
+            "label": "‮",
+        }
+        "ALGR(FA_HEJ)": {
+            "key": "FA_PDIR",
+            "label": "‬",
+        }
+        "ALGR(FA_JIM)": {
+            "key": "FA_LTRE",
+            "label": "‪",
+        }
+        "ALGR(FA_CHE)": {
+            "key": "FA_RTLE",
+            "label": "‫",
+        }
+        "ALGR(FA_YE)": {
+            "key": "FA_ALFM",
+            "label": "ى",
+        }
+        "ALGR(FA_ALEF)": {
+            "key": "FA_ALFW",
+            "label": "ٱ",
+        }
+        "ALGR(FA_NOON)": {
+            "key": "FA_LORP",
+            "label": "﴾",
+        }
+        "ALGR(FA_MIM)": {
+            "key": "FA_RORP",
+            "label": "﴿",
+        }
+        "ALGR(FA_KAF)": {
+            "key": "FA_ESCN",
+            "label": ";",
+        }
+        "ALGR(FA_GAF)": {
+            "key": "FA_DQT",
+            "label": "\"",
+        }
+        "ALGR(FA_BSLS)": {
+            "key": "FA_MNSA",
+            "label": "-",
+        }
+        "ALGR(FA_ZA)": {
+            "key": "FA_PIPE",
+            "label": "|",
+        }
+        "ALGR(FA_RA)": {
+            "key": "FA_SUBA",
+            "label": "ٖ",
+        }
+        "ALGR(FA_ZAL)": {
+            "key": "FA_JSPA",
+            "label": "‍",
+        }
+        "ALGR(FA_DAL)": {
+            "key": "FA_HMZB",
+            "label": "ء",
+        }
+        "ALGR(FA_PE)": {
+            "key": "FA_ELLP",
+            "label": "…",
+        }
+        "ALGR(FA_VAV)": {
+            "key": "FA_ECOM",
+            "label": ",",
+        }
+        "ALGR(FA_DOT)": {
+            "key": "FA_EQOT",
+            "label": "'",
+        }
+        "ALGR(FA_SLSH)": {
+            "key": "FA_EQUS",
+            "label": "?",
+        }
+/* Shift+AltGr symbols
+ * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───────┐
+ * │   │ 1 │ 2 │ 3 │ 4 │ 5 │ 6 │ 7 │ 8 │ 9 │ 0 │   │   │       │
+ * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─────┤
+ * │     │   │   │   │   │   │   │   │   │   │   │   │   │     │
+ * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┐    │
+ * │      │   │   │   │   │   │   │   │   │   │   │   │   │    │
+ * ├────┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┴────┤
+ * │    │ ¦ │   │   │   │   │   │   │   │   │   │   │          │
+ * ├────┼───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤
+ * │    │    │    │                        │    │    │    │    │
+ * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+ */
+        "S(ALGR(FA_1))": {
+            "key": "FA_1E",
```

Comment:
These should be swapped around.

Reply:
Can you explain more? Compared to other languages this looks fine.

Reply:
`FA_1` should print `1` etc, not the Eastern Arabic numerals. Same goes for a few other keycodes.

Reply:
The standard Farsi (Persian) keyboard offers Persian numerals on the first layout which are at the same position of their English numerals. I think when I'm defining a keyboard layout for this language I want to refer to `1` as it's corresponding Persian one which is `۱` etc.

Reply:
That's not really how these keycodes are supposed to be named.

Reply:
But that makes much more sense to me. If I understand the purpose of extra keymaps correctly, they are used to define keymaps based on other languages, so when I'm defining a custom keymap in Farsi, most of the time I want the Farsi numerals and English version is the alternative one, so I use the `FA_1` to represent `۱`.

If there is a convention to not write the keymaps this way, I couldn't find it anywhere in other languages. Can you give me an example? I think there are no other languages that currently have this problem.

Reply:
The example is everything else. `FA_1` should print `1`, just as `KC_1` prints `1`. `1` and `۱` are *not* the same character, though they may mean the same thing.

Reply:
Everything else is not what you said. For example:

https://github.com/qmk/qmk_firmware/blob/2fd56317763e8b3b73f0db7488ef42a70f5b946e/data/constants/keycodes/extras/keycodes_russian_0.0.1.hjson#L76-L79

In Russian `RU_U` doesn't print `u`.

Reply:
That is different because we are only using ASCII characters for keycode names, otherwise it would indeed be `RU_У`.
As you can see from the surrounding keycodes there, the convention is to derive the names from the Unicode name, eg:
![image](https://github.com/qmk/qmk_firmware/assets/4781841/3552688d-beb3-456d-b238-09ad04d6f472)

I guess for these I would just call them `FA_nA`:
![image](https://github.com/qmk/qmk_firmware/assets/4781841/66178ae0-a930-4154-861c-e63e72d293fd)

Reply:
OK, that makes sense (although the convention is not good IMHO).
I'm changing these, but I have another question, what should I do with repeated keys? There are certain keys that appear multiple times, for example `200c` unicode character appears multiple times in the `Shift` layer. Previously I defined them multiple times with different names.